src/onemod/config/base.py

```diff
@@ -1,156 +1,124 @@
-        cov_name: (lower, upper). Default is None.
+        return key in self._get_fields()
 
-    """
+    def __repr__(self) -> str:
+        arg_list = []
+        for key in self._get_fields():
+            arg_list.append(f"{key}={getattr(self, key)!r}")
+        arg_list.sort()
+        return f"{type(self).__name__}({', '.join(arg_list)})"
 
-    id_columns: set[str]
-    model_type: Literal["binomial", "gaussian", "poisson"]
-    observation_column: str = "obs"
-    prediction_column: str = "pred"
-    weights_column: str = "weights"
-    test_column: str = "test"
-    holdout_columns: set[str] | None = None
-    coef_bounds: dict[str, tuple[float, float]] | None = None
+    def _get_fields(self) -> list[str]:
+        return list(self.model_dump(exclude_none=True))
 
 
 class StageConfig(Config):
     """Stage configuration class.
 
-    If None, setting inherited from Pipeline.
-
-    Attributes
-    ----------
-    id_columns : set[str] or None, optional
-        ID column names, e.g., 'age_group_id', 'location_id', 'sex_id',
-        or 'year_id'. ID columns should contain nonnegative integers.
-        Default is None.
-    model_type : str or None, optional
-        Model type; either 'binomial', 'gaussian', or 'poisson'.
-        Default is None.
-    observation_column : str or None, optional
-        Observation column name for pipeline input. Default is None.
-    prediction_column : str or None, optional
-        Prediction column name for pipeline output. Default is None.
-    weights_column : str or None, optional
-        Weights column name for pipeline input. The weights column
-        should contain nonnegative floats. Default is None.
-    test_column : str or None, optional
-        Test column name. The test column should contain values 0
-        (train) or 1 (test). The test set is never used to train stage
-        models, so it can be used to evaluate out-of-sample performance
-        for the entire pipeline. If no test column is provided, all
-        missing observations will be treated as the test set. Default is
-        None.
-    holdout_columns : set[str] or None, optional
-        Holdout column names. The holdout columns should contain values
-        0 (train), 1 (holdout), or NaN (missing observations). Holdout
-        sets are used to evaluate stage model out-of-sample performance.
-        Default is None.
-    coef_bounds : dict or None, optional
-        Dictionary of coefficient bounds with entries
-        cov_name: (lower, upper). Default is None.
+    If a StageConfig instance does not contain an attribute, the get and
+    __getitem__ methods will return the corresponding pipeline
+    attribute, if it exists.
 
     """
 
-    model_config = ConfigDict(extra="allow")
+    model_config = ConfigDict(
+        extra="allow", validate_assignment=True, protected_namespaces=()
+    )
+
+    _pipeline_config: Config = Config()
+    _required: set[str] = set()  # TODO: unique list
```

Comment:
Noted - will be sure to address these when I ultimately merge these changes into my UniqueList branch